auth/allow_all_authenticator.hh

```diff
@@ -88,11 +88,6 @@ public:
         return make_ready_future<custom_options>();
     }
 
-    virtual const resource_set& protected_resources() const override {
-        static const resource_set resources;
-        return resources;
-    }
-
     virtual ::shared_ptr<sasl_challenge> new_sasl_challenge() const override {
```

Comment:
We could have retained it, by making it return a protector (or vector<protector>) instead of a resource_set.

But registering a protector is nicer in a way. Btw, I think you forgot to unregister it on stop.

Reply:
(it's nicer in that the enforcer doesn't have to name the services it protects, instead the services that wish protection have to enlist)

Reply:
On second thought, registering and unregistering protectors correctly is a lot of extra code, and I'm not convinced we quite need it yet.  I decided to adapt existing is_protected to work with command_desc.